contracts/exchange-forwarder/test/asset_test.ts

```diff
@@ -115,8 +126,49 @@ blockchainTests('Supported asset type unit tests', env => {
                 .callAsync();
             expect(result).to.be.false();
         });
-        it('returns false if assetData1 == assetData2 are ERC721', async () => {
+        it('returns true if assetData1 == assetData2 are ERC721', async () => {
             const result = await forwarder.areUnderlyingAssetsEqual(erc721AssetData, erc721AssetData).callAsync();
+            expect(result).to.be.true();
+        });
+        it('returns false if assetData1 != assetData2 are ERC721', async () => {
+            const differentErc721AssetData = assetDataEncoder
+                .ERC721Token(randomAddress(), getRandomInteger(0, constants.MAX_UINT256))
+                .getABIEncodedTransactionData();
+            const result = await forwarder
+                .areUnderlyingAssetsEqual(erc721AssetData, differentErc721AssetData)
+                .callAsync();
+            expect(result).to.be.false();
+        });
+        it('returns true if assetData1 == assetData2 are StaticCall', async () => {
+            const result = await forwarder
+                .areUnderlyingAssetsEqual(staticCallAssetData, staticCallAssetData)
+                .callAsync();
+            expect(result).to.be.true();
+        });
+        it('returns false if assetData1 != assetData2 are StaticCall', async () => {
+            const differentStaticCallAssetData = assetDataEncoder
+                .StaticCall(randomAddress(), hexUtils.random(), constants.KECCAK256_NULL)
+                .getABIEncodedTransactionData();
+            const result = await forwarder
+                .areUnderlyingAssetsEqual(staticCallAssetData, differentStaticCallAssetData)
+                .callAsync();
+            expect(result).to.be.false();
+        });
+        it('returns false if assetData1 is ERC20 and assetData2 is MultiAsset', async () => {
+            const result = await forwarder.areUnderlyingAssetsEqual(erc20AssetData, multiAssetData).callAsync();
+            expect(result).to.be.false();
+        });
+        it('returns true if assetData1 == assetData2 are MultiAsset', async () => {
+            const result = await forwarder.areUnderlyingAssetsEqual(multiAssetData, multiAssetData).callAsync();
+            expect(result).to.be.true();
+        });
+        it('returns false if assetData1 != assetData2 are MultiAsset', async () => {
+            const differentMultiAssetData = assetDataEncoder
+                .MultiAsset([getRandomInteger(0, constants.MAX_UINT256)], [erc721AssetData])
+                .getABIEncodedTransactionData();
+            const result = await forwarder
+                .areUnderlyingAssetsEqual(multiAssetData, differentMultiAssetData)
+                .callAsync();
```

Comment:
Could be worthwhile to validating equality of `MultiAsset` with multiple nested assets.